code/game/objects/items/weapons/melee/misc.dm

```diff
@@ -106,3 +106,196 @@
+	enchant = E
+	E.on_gain(src, user)
+	E.power *= power
+	if(intentional)
+		SSblackbox.record_feedback("nested tally", "spellblade_enchants", 1, list("[E.name]"))
+
+/obj/item/melee/spellblade/examine(mob/user)
+	. = ..()
+	if(enchant && (iswizard(user) || iscultist(user)))
+		. += "The runes along the side read; [enchant.desc]."
+
+
+/obj/item/melee/spellblade/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(attack_type == PROJECTILE_ATTACK)
+		final_block_chance = 0
+	return ..()
+
+/datum/enchantment
+	/// used for blackbox logging
+	var/name = "You shouldn't be seeing this, file an issue report."
+	/// used for wizards/cultists examining the runes on the blade
+	var/desc = "Someone messed up, file an issue report."
+	/// used for damage values
+	var/power = 0
+	/// whether the enchant procs despite not being in proximity
+	var/ranged = FALSE
+	/// stores the world.time until it can be used again, the `initial(cooldown)` is the cooldown between activations.
+	var/cooldown = -1
+
+/datum/enchantment/proc/on_hit(mob/living/target, mob/living/user, proximity, obj/item/melee/spellblade/S)
+	if(world.time < cooldown)
+		return FALSE
+	if(!istype(target))
+		return FALSE
+	if(target.stat == DEAD)
+		return FALSE
+	if(!ranged && !proximity)
+		return FALSE
+	cooldown = world.time + initial(cooldown)
+	return TRUE
+
+/datum/enchantment/proc/on_gain(obj/item/melee/spellblade, mob/living/user)
+
+/datum/enchantment/lightning
+	name = "lightning"
+	desc = "this blade conducts arcane energy to arc between its victims"
+	power = 20
+	cooldown = 3 SECONDS
+
+/datum/enchantment/lightning/on_hit(mob/living/target, mob/living/user, proximity, obj/item/melee/spellblade/S)
+	. = ..()
+	if(.)
+		zap(target, user, list(user), power)
+
+
+/datum/enchantment/lightning/proc/zap(mob/living/target, mob/living/source, protected_mobs, voltage)
+	source.Beam(target, "lightning[rand(1,12)]", 'icons/effects/effects.dmi', time = 2 SECONDS, maxdistance = 7, beam_type = /obj/effect/ebeam/chain)
+	if(!target.electrocute_act(voltage, flags = SHOCK_TESLA)) // if it fails to shock someone, break the chain
+		return
+	protected_mobs += target
+	addtimer(CALLBACK(src, .proc/arc, target, voltage, protected_mobs), 2.5 SECONDS)
+
+/datum/enchantment/lightning/proc/arc(mob/living/source, voltage, protected_mobs)
+	voltage = voltage - 4
+	if(voltage <= 0)
+		return
+
+	for(var/mob/living/L in oview(7, get_turf(source)))
+		if(L in protected_mobs)
+			continue
+		zap(L, source, protected_mobs, voltage)
+		break
+
+/datum/enchantment/fire
+	name = "fire"
+	desc = "this blade will self immolate on hit, releasing a ball of fire. it also makes the weilder immune to fire"
+	power = 20
+	cooldown = 8 SECONDS
+
+/datum/enchantment/fire/on_gain(obj/item/melee/spellblade/S, mob/living/user)
+	..()
+	RegisterSignal(S, list(COMSIG_ITEM_PICKUP, COMSIG_ITEM_DROPPED), .proc/toggle_traits)
```

Comment:
This could theoretically cause issues if you have two spell swords with the fire enchantment.
Grab one, get traits. Grab another, lose traits.

Reply:
This is also gonna cause issues with the eternal flame spell. I knew i was forgetting something